lib/seahorn/HornifyModule.cc

```diff
@@ -92,6 +98,27 @@ static llvm::cl::list<std::string>
                       llvm::cl::desc("Abstract all calls to these functions"),
                       llvm::cl::ZeroOrMore, llvm::cl::CommaSeparated);
 
+static llvm::cl::opt<bool, true> InterProcMem(
+    "horn-inter-proc-mem",
+    llvm::cl::desc("Use inter-procedural encoding with memory"), llvm::cl::init(false));
+
+namespace seahorn {
+// counters for copying the new inter-proc vcgen
+// only updated if the log "inter_mem_counters" is active
```

Comment:
these seem like global variables. So names should be `g_`. 

However, this brings another question, why do we need global variables? Why can't they be members of `HornifyModule`?

Reply:
The `HornifyModule` class is not available inside of `UfoOpSem`, and the counting takes place there.

Ideally, they should not be accessed in `HornifyModule`. However, the processing done in `UfoOpSem` is done twice for each callsite in `runOnFunction` because of `r.first->second.run();` and `hf->runOnFunction(F);`, the counters are counted twice if we don't do this.

Reply:
I don't understand the answer. If they are only modified by `UfoOpSem`, they should be defined there. If they need to be reset from `HornifyModule`, then `UfoOpSem` can provide API to reset the counters. In any case, the name should change to reflect that they are global, if they are to stay like that for now.